src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6455,4 +6461,24 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_FeatureExtensionGetEnumerator" xml:space="preserve">
     <value>extension GetEnumerator</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="ERR_UnmanagedCallersOnlyRequiresStatic" xml:space="preserve">
+    <value>'UnmanagedCallersOnly' can only be applied to ordinary static methods or static local functions.</value>
+    <comment>UnmanagedCallersOnly is not localizable.</comment>
+  </data>
+  <data name="ERR_InvalidUnmanagedCallersOnlyCallConv" xml:space="preserve">
+    <value>'{0}' is not a valid calling convention type for 'UnmanagedCallersOnly'.</value>
+    <comment>UnmanagedCallersOnly is not localizable.</comment>
+  </data>
+  <data name="ERR_CannotUseManagedTypeInUnmanagedCallersOnly" xml:space="preserve">
+    <value>Cannot use '{0}' as a {1} type on a method attributed with 'UnmanagedCallersOnly'.</value>
+    <comment>1 is the localized word for 'parameter' or 'return'. UnmanagedCallersOnly is not localizable.</comment>
+  </data>
+  <data name="ERR_UnmanagedCallersOnlyMethodCannotBeGeneric" xml:space="preserve">
+    <value>Methods attributed with 'UnmanagedCallersOnly' cannot be generic.</value>
```

Comment:
>Methods attributed with 'UnmanagedCallersOnly' cannot be generic. [](start = 11, length = 65)

`ERR_UnmanagedCallersOnlyRequiresStatic` is reported on the attribute but `ERR_UnmanagedCallersOnlyMethodCannotBeGeneric` is reported on the method. Consider reporting both on the attribute.